tests/components/shelly/test_init.py

```diff
@@ -309,3 +310,25 @@ async def test_entry_missing_gen(hass: HomeAssistant, mock_block_device) -> None
 
     assert entry.state is ConfigEntryState.LOADED
     assert hass.states.get("switch.test_name_channel_1").state is STATE_ON
+
+
+@pytest.mark.parametrize(("entry_sleep", "device_sleep"), [(21600, 42300)])
+async def test_sleeping_block_device_wrong_sleep_period(
+    hass: HomeAssistant,
+    entry_sleep,
+    device_sleep,
+    mock_block_device,
+    caplog: pytest.LogCaptureFixture,
+) -> None:
+    """Test sleeping block device with wrong sleep period."""
+
+    await init_integration(hass, 1, model=MODEL_DW, sleep_period=entry_sleep)
+    assert "will resume when device is online" in caplog.text
+
+    mock_block_device.mock_update()
+    await hass.async_block_till_done()
+
+    assert (
+        f"Updating stored sleep period for Mock Title: from {entry_sleep} to {device_sleep}"
+        in caplog.text
+    )
```

Comment:
Example for test which verify the value is updated instead of checking the log:
```suggestion
async def test_sleeping_block_device_wrong_sleep_period(
    hass: HomeAssistant, mock_block_device
) -> None:
    """Test sleeping block device with wrong sleep period."""

    entry = await init_integration(
        hass, 1, model=MODEL_DW, sleep_period=21600, skip_setup=True
    )
    assert entry.data[CONF_SLEEP_PERIOD] == 21600

    await hass.config_entries.async_setup(entry.entry_id)
    await hass.async_block_till_done()

    assert entry.data[CONF_SLEEP_PERIOD] == 43200
```

If the 21600/43200 are moved to constants they can be used here instead
We can also replace the `MODEL_DW` with a parameter if the model list is also a constant and let it run over the list using `@pytest.mark.parametrize(("model"), BLOCK_MODELS_WITH_WRONG_SLEEP_PERIOD)`

Reply:
Test log 👍 
```
  File "/home/vscode/.local/lib/python3.11/site-packages/aioshelly/block_device/device.py", line 275, in _coap_request
    await self.coap_context.request(self.ip_address, path)
TypeError: object Mock can't be used in 'await' expression
```

Reply:
You are missing `mock_block_device` in your fixture, I run the test suggested on your changes and they work (assuming the sleep period is 43200 and not 42300 (but in any case there is no error)